lib/mini_twitter_client.rb

```diff
@@ -0,0 +1,81 @@
+require 'faraday'
+require 'json'
+
+class Tweet
+  def initialize(name = 'name', email = 'email@email.com', message = 'message', id = nil)
+    @name = name
+    @email = email
+    @message = message
+    @id =id
+  end
+  def name
+    @name
+  end
+  def email
+    @email
+  end
+  def message
+    @message
+  end
+  def id
+    @id
+  end
+end
+
+class MiniTwitterClient
+  def initialize
+    @connection = Faraday.new(
+      url: 'http://localhost:4000',
+      params: {},
+      headers: {
+        'Accept' => 'application/json',
+        'Content-Type' => 'application/json'
+      }
+    )
+  end
+
+  def get_tweets
+    res = @connection.get('/api/tweets')
+    data = JSON.parse(res.body)['data'].map do |tweet|
+      Tweet.new(tweet['author']['name'], tweet['author']['email'], tweet['message'], tweet['id'])
+    end
+
+    {
+      status: res.status.to_i,
+      data:
+    }
+  end
+
+  def create_tweet(name, email, message)
+    res = @connection.post('/api/tweets') do |post|
+      post.body = { "tweet": { "author": { "name": name, "email": email }, "message": message } }.to_json
+    end
+    tweet = JSON.parse(res.body)['data']
+
+    {
+      status: res.status.to_i,
+      data: Tweet.new(tweet['author']['name'], tweet['author']['email'], tweet['message'], tweet['id'])
+    }
+  end
+
+  def delete_tweet(id)
+    res = @connection.delete("/api/tweets/#{id}")
+
+    {
+      status: res.status.to_i,
+      data: Tweet.new()
```

Comment:
Sending Tweet with default data would be misleading here. Since the endpoint responds with empty body, I would just set data to `nil`:

```suggestion
      data: nil
```

Or just remove the data completely:

```suggestion
```

Reply:
`data: nil` it is ok for me, I like to keep format of response. Maybe in this example is not important, but still like to keep it.